tests/transforms/test_draw.py

```diff
@@ -1,593 +1,242 @@
-            return qml.expval(qml.PauliX(wires=[0]) @ qml.PauliX(wires=[1]) @ qml.PauliX(wires=[2]))
-
-        res = qml.draw(qnode)()
-        expected = [
-            " 0: ──╭┤ ⟨X ⊗ X ⊗ X⟩ \n",
-            " 1: ──├┤ ⟨X ⊗ X ⊗ X⟩ \n",
-            " 2: ──╰┤ ⟨X ⊗ X ⊗ X⟩ \n",
-        ]
-
-        assert res == "".join(expected)
-
-
-class TestOpsIntegration:
-    """Integration tests for drawing specific operations and templates"""
-
-    def test_approx_time_evolution(self):
-        """Test that a QNode with the ApproxTimeEvolution template draws
-        correctly when having the expansion strategy set."""
-        H = qml.PauliX(0) + qml.PauliZ(1) + 0.5 * qml.PauliX(0) @ qml.PauliX(1)
-
-        @qml.qnode(qml.device("default.qubit", wires=2))
-        def circuit(t):
-            qml.ApproxTimeEvolution(H, t, 2)
-            return qml.probs(wires=0)
-
-        res = qml.draw(circuit, expansion_strategy="device")(0.5)
-        expected = [
-            " 0: ──H────────RZ(0.5)──H──H──╭RZ(0.25)──H──H────────RZ(0.5)──H──H──╭RZ(0.25)──H──┤ Probs \n",
-            " 1: ──RZ(0.5)──H──────────────╰RZ(0.25)──H──RZ(0.5)──H──────────────╰RZ(0.25)──H──┤       \n",
-        ]
-
-        assert res == "".join(expected)
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Integration tests for the draw transform
+"""
+import pytest
+from functools import partial
+
+import pennylane as qml
+from pennylane import numpy as np
+
+from pennylane.transforms import draw
+
+
+@qml.qnode(qml.device("default.qubit", wires=(0, "a", 1.234)))
+def circuit(x, y, z):
+    qml.RX(x, wires=0)
+    qml.RY(y, wires="a")
+    qml.RZ(z, wires=1.234)
+    return qml.expval(qml.PauliZ(0))
+
+
+class TestLabelling:
+    """Test the wire labels."""
+
+    def test_any_wire_labels(self):
+        """Test wire labels with different kinds of objects."""
+
+        split_str = draw(circuit)(1.2, 2.3, 3.4).split("\n")
+        assert split_str[0][0:6] == "    0:"
+        assert split_str[1][0:6] == "    a:"
+        assert split_str[2][0:6] == "1.234:"
+
+    def test_wire_order(self):
+        """Test wire_order keyword changes order of the wires."""
+
+        split_str = draw(circuit, wire_order=[1.234, "a", 0, "b"])(1.2, 2.3, 3.4).split("\n")
+        assert split_str[0][0:6] == "1.234:"
+        assert split_str[1][0:6] == "    a:"
+        assert split_str[2][0:6] == "    0:"
+
+    def test_show_all_wires(self):
```

Comment:
Could this test include the case where extra wires are present from the wire_order argument?

Reply:
added below.